### PR TITLE
Add support for PerfEvents

### LINKED ISSUE #13: Perf events was supported?
Hi,

The library is awesome!  Thanks for great work. 

But I wonder whether it already supported BPF_PERF_EVENT_OUTPUT() helper?

### PR DESCRIPTION
**Pull request by belyalov**
Adding support for most common, index to `cpu_id` perfEvents support.

Fixes #13

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -73,5 +73,51 @@ Once compiled can be used by goebpf in the following way:
...
+	defer program.Detach()
+
+	// Start listening to Perf Events
+	perf, err := goebpf.NewPerfEvents(perfmap)
+	// 4096 is ring buffer size
+	perfEvents, err := perf.StartForAllProcessesAndCPUs(4096)
+	defer perf.Stop()
+
+	for {
+		switch {
```
[the remark this answers is not in the document]
Comment (belyalov) [reply doc.go]: Yeah, stupid typo.. ) Fixed

**Comment by belyalov** [comment]
Integration test will be added in next PR (current is big enough)

File: perf_events.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	EventsLost       int
+	EventsUnknowType int
+
+	// PollTimeoutMs is timeout for blocking call of poll()
+	// Defaults to 100ms
+	PollTimeoutMs int
+
+	perfMap        Map
+	updatesChannel chan []byte
+	stopChannel    chan bool
```
Comment (veshij) [inline perf_events.go]: nit: as this chan is used only for signalling you might want to use `stopChannel chan struct{}` (0-sized object)

File: perf_events.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+// All updates will be sent into returned channel.
+func (pe *PerfEvents) StartForAllProcessesAndCPUs(bufferSize int) (<-chan []byte, error) {
+	// Get CPU count
+	nCpus, err := GetNumOfPossibleCpus()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create perfEvent handler for all possible CPUs
+	pe.handlers = nil
```
Comment (veshij) [inline perf_events.go]: nit: `pe.handlers = make([]*.., nCpus)` to avoid allocations in loop.

File: perf_events_poller.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    // Re-set pointers to start of arrays
+    pollfds = pollfds_memory;
+    fds = _fds;
+
+    int ready_cnt = poll(pollfds, cnt, timeout);
+
+    // Copy all ready descriptors back into golang array of uint32s
+    for (int remain = ready_cnt; remain > 0; pollfds++) {
+        if (pollfds->revents & POLLIN) {
```
Comment (tehnerd) [inline perf_events_poller.go]: from man:
" The bits returned in revents can
       include any of those specified in events, or one of the values
       POLLERR, POLLHUP, or POLLNVAL. "

so e.g. if there was any event which is not POLLIN we would never have remain == 0 and would try to access thinigs outside pollfd's allocated memory (w/ eventual segfolt)

File: perf_events.go (hunk 2 above)
Comment (belyalov) [reply perf_events.go]: I don't have any preference here, so changed to `chan struct{}` :)

File: perf_events.go (hunk 3 above)
Comment (belyalov) [reply perf_events.go]: This is one time call function so I'd consider this is as minor.
Anyway refactored to make it more readable.

File: perf_events_poller.go (hunk 4 above)
Comment (belyalov) [reply perf_events_poller.go]: Good catch, added one more condition to avoid this:
```
pollfds != pollfds_end
```

**Pull request merged**